distributed/scheduler.py

```diff
@@ -4566,10 +4571,14 @@ def _add_to_memory(
         if ts in cs._wants_what:
             self.client_releases_keys(client="fire-and-forget", keys=[ts._key])
```

Comment:
This seems like something we need to move out of here as well, but am less sure how we do that.

Reply:
Have tried to address this by collecting the `recommendations` from `client_releases_keys` and passing those along back to the caller to in turn return back to `transition` to handle.